tutorials/hoon/landscape-tile.md

```diff
@@ -0,0 +1,276 @@
+  (send-status-diff "failure")
+++  poke-handle-http-request
+  %-  (require-authorization:app ost.bol move .)
+  |=  =inbound-request:eyre
+  ^-  (quip move _+>.$)
+  =/  request-line  (parse-request-line url.request.inbound-request)
+  =/  back-path  (flop site.request-line)
+  =/  name=@t
+    =/  back-path  (flop site.request-line)
+    ?~  back-path
+      ''
+    i.back-path
+  ?~  back-path
+    [[ost.bol %http-response not-found:app]~ +>.$]
+  ?:  =(name 'tile')
+    [[ost.bol %http-response (js-response:app tile-js)]~ +>.$]
+  [[ost.bol %http-response not-found:app]~ +>.$]
+::
+--
+```
+
+Now, replace the code below is the contents of `/app/testing/js/tile.js` with the code below. Once you've replaced the code in both files, run `|commit %home` in the Dojo.
+
+``` javascript
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import _ from 'lodash';
+
+
+export default class testingTile extends Component {
+
+    constructor(props) {
+        super(props);
+        console.log("og props");
+        console.log(this.props);
+        this.state = { ship: "~zod" };
+    }
+
+    sub() {
+        api.action('testing', 'json', {ship: this.state.ship});
+    }
+
+    handleChange(event) {
+        this.setState({ship: event.target.value});
+    }
+
+    render() {
+    return (
+      <div className="w-100 h-100 relative" style={{ background: '#1a1a1a' }}>
+          <p className="gray label-regular b absolute" style={{ left: 8, top: 4 }}>Testing</p>
+            <p className="white absolute" style={{ top: 25, left: 8 }}>Hi a ship {this.props.data.status}</p>
+            <p className="white absolute" style={{ top: 100, left: 8 }}>
+            <button onClick={this.sub.bind(this)}>stuff</button></p>
+            <p className="white absolute" style={{ top: 150, left: 15 }}>
+            <input type="text" value={this.state.ship}
+             onChange={this.handleChange.bind(this)}/></p>
+      </div>
+    );
+  }
+
+}
+
+window.testingTile = testingTile;
+```
+
+### Default Code Components
+
+Parts of the `testing.hoon` code shown above is the "scaffolding" that was automatically built for us by the default Hoon file from the `create-landscape-app` repo. We'll give a general overview of those arms here.
+
+As with other gall apps, `card` and `move` define the types that can be produced by the app, `card` specifically defines which type of requests can be made to other parts of the system.
+
+`++prep` is called when the application first starts, and when its code gets updated. It will get a `unit` of the old state and `++prep` needs to update the state. You might notice the use of `~&`. This is useful to give us feedback to make sure our changes have been loaded.
+
+The `++peer-testingtile` arm is called when the tile first subscribes to the app. The subscription logic is mostly handled for us by the `launch` app here it simply produces some data to send back to the tile, indicating its initial state.
```

Comment:
What specifies the subscription being opened, is it keyed off the `testingTile` class name? Also, does "initial `state`" here mean `props.data`, in react terms?